### ISSUE TITLE
Bad Vulkan data?

### ISSUE THREAD

**Issue by jarredwalton** [GameTechDev/PresentMon#14]
I've tried using PresentMon with two different Vulkan games, The Talos Principle and now Doom (which just got a Vulkan patch today). I didn't realize the problem on Talos, but it appears looking at the CSV files that PresentMon is capturing 'MsBetweenPresents' incorrectly or something -- or maybe Vulkan is just doing things wrong? Anyway, looking at the CSV files, all of the 'Hardware: Legacy Flip' frames have an interval that's a multiple of the display refresh time (e.g. 16.67 ms for 60 Hz).

If you manually subtract the previous row time from the current row time (column 'TimeInSeconds'), you get completely different numbers -- and some of the numbers are negative! So this method of determining frame times (and fps) is also borked.

In other words, right now with both publicly available Vulkan games, PresentMon is not providing useful data AFAICT. Thanks!

**Comment by jenatali** [comment]
Can you share an example CSV?

**Comment by jarredwalton** [comment]
Sure... here you go. I don't have 'simple' files for Talos, but I do have both 'simple' and 'standard' files for Doom in the 7-zip archive. Note that (1) files are 1080p, (2) is 1440p, and (3) is 4K. Thanks!
--Jarred Walton (PC Gamer)

[PresentMon Vulkan Benchmarks.zip](https://github.com/GameTechDev/PresentMon/files/358018/PresentMon.Vulkan.Benchmarks.zip)

**Comment by jenatali** [comment]
You should use the -process_name command line flag (e.g. "-process_name DOOMx64vk.exe"). The ones with the 16.67ms are from a different process, probably dwm.exe.

**Comment by jarredwalton** [comment]
Ah, poop -- that would explain it. Strangely, this doesn't seem to affect DX12 games, only Vulkan (and probably Mantle). Targeting a specific process (e.g. for Doom) seems to have fixed the problem, but that means I now need to always pass the process to PresentMon -- not super convenient when you're testing a bunch of different games, but such is life I guess.

**Comment by jenatali** [comment]
Yeah, PresentMon is system-wide. If you're running a game in fullscreen, then DWM stops presenting. For DX12 games, this can even include borderless windowed mode. I'm unsure if Vulkan currently supports fullscreen, but I know that its borderless windowed mode still requires DWM to run.

You can also just post-process the CSV by filtering based on process ID or process name. The columns should be coherent for each process if you do that, but by default they're sorted based on time, not process.
